src/main/java/com/synopsys/integration/detect/tool/binaryscanner/BinaryUploadOperation.java

```diff
@@ -66,6 +66,26 @@ public CodeLocationCreationData<BinaryScanBatchOutput> uploadBinaryScanFile(
             throw new DetectUserFriendlyException("Failed to upload binary scan file.", e, ExitCodeType.FAILURE_BLACKDUCK_FEATURE_ERROR);
         }
     }
+    
+    public CodeLocationCreationData<BinaryScanBatchOutput> uploadBinaryScanFiles(
+        BinaryScanBatch binaryScanBatch,
+        BinaryScanUploadService binaryScanUploadService,
+        NameVersion projectNameVersion
+    )
+        throws DetectUserFriendlyException {
+        try {
+            CodeLocationCreationData<BinaryScanBatchOutput> codeLocationCreationData = binaryScanUploadService.uploadBinaryScan(binaryScanBatch);
+
+            BinaryScanBatchOutput binaryScanBatchOutput = codeLocationCreationData.getOutput();
+            // The throwExceptionForError() in BinaryScanBatchOutput has a bug, so doing that work here
+            throwExceptionForError(binaryScanBatchOutput);
+            statusEventPublisher.publishStatusSummary(new Status(STATUS_KEY, StatusType.SUCCESS));
+            return codeLocationCreationData;
+        } catch (IntegrationException e) {
+            statusEventPublisher.publishStatusSummary(new Status(STATUS_KEY, StatusType.FAILURE));
+            throw new DetectUserFriendlyException("Failed to upload binary scan file.", e, ExitCodeType.FAILURE_BLACKDUCK_FEATURE_ERROR);
+        }
+    }
```

Comment:
I understand the intent of writing this new method even though we have `BinaryUploadOperation.uploadBinaryScanFile` which does something similar. It would have been easier if that method was better written to just accept a batch but I guess that is a tech-debt for us to handle.

Reply:
The two methods should not be combined before we understand the nuances of binary scan handling between a single request and a batch.

Reply:
Yeah that is a fair point for now, the existing original method could have been more generic. We can revisit this before merging to master.